commit 17b8532a71b005a075fa7051fb9a9b32bdf944a2
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Tue Feb 28 22:34:31 2023 -0500
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Feb 28 19:34:31 2023 -0800

    Enhance the job run process not to kill its own process, instead let … (#1440)
    
    * Enhance the job run process not to kill its own process, instead let it to MPM manage.
    
    * refactored.
    
    ---------
    
    Co-authored-by: Yuan-Ting Hsieh (謝沅廷) <yuantingh@nvidia.com>
---
 nvflare/private/fed/server/server_engine.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index b7dc502d..4e308f1e 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -491,6 +491,7 @@ class ServerEngine(ServerEngineInternalSpec):
             channel=CellChannel.SERVER_PARENT_LISTENER,
             topic=ServerCommandNames.GET_CLIENTS,
             request=request,
+            timeout=5.0,
         )
         data = fobs.loads(return_data.payload)
         if data.get(ServerCommandKey.CLIENTS):
@@ -514,13 +515,15 @@ class ServerEngine(ServerEngineInternalSpec):
                 message=request,
             )
 
-    def send_command_to_child_runner_process(self, job_id: str, command_name: str, command_data, return_result=True):
+    def send_command_to_child_runner_process(
+        self, job_id: str, command_name: str, command_data, return_result=True, timeout=5.0
+    ):
         result = None
         with self.lock:
             fqcn = FQCN.join([FQCN.ROOT_SERVER, job_id])
             request = new_cell_message({}, fobs.dumps(command_data))
             return_data = self.server.cell.send_request(
-                target=fqcn, channel=CellChannel.SERVER_COMMAND, topic=command_name, request=request
+                target=fqcn, channel=CellChannel.SERVER_COMMAND, topic=command_name, request=request, timeout=timeout
             )
             result = fobs.loads(return_data.payload)
 
